organisations/models.py

```diff
@@ -23,6 +25,30 @@ def user_can_view(self, user):
         else:
             return self.none()
 
+    def retention_period_exceeded(self):
+        earliest_valid_timestamp = timezone.now() - relativedelta(
+            months=settings.PERSONAL_DATA_RETENTION_PERIOD_MONTHS
+        )
+
+        all_enrolments_too_old = Q(max_enrolment_end_time=None) | Q(
+            max_enrolment_end_time__lt=earliest_valid_timestamp
+        )
+        all_study_groups_too_old = Q(max_studygroup_end_time=None) | Q(
+            max_studygroup_end_time__lt=earliest_valid_timestamp
+        )
+
+        return (
+            # At least for now returns only unauthenticated Person's
+            self.filter(user=None)
+            .annotate(
+                max_enrolment_end_time=Max("enrolment__occurrence__end_time"),
+                max_studygroup_end_time=Max(
+                    "studygroup__enrolments__occurrence__end_time"
```

Comment:
another one is in a plural form? "enrolments"?

Reply:
The first one comes from `Enrolment` -> `Person` relation, where `related_name` is not set, so it defaults to the model's name. The second one comes from `Enrolment` -> `StudyGroup` relation, where `related_name` is set to `enrolments`.